src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/Security/Factory/AbstractFactoryTest.php

```diff
@@ -145,7 +145,12 @@ public static function getSuccessHandlers()
 
     protected function callFactory($id, $config, $userProviderId, $defaultEntryPointId)
     {
-        $factory = $this->getMockForAbstractClass(AbstractFactory::class);
+        $factory = $this->getMockBuilder(AbstractFactory::class)->onlyMethods([
+            'createAuthProvider',
+            'getListenerId',
+            'getKey',
+            'getPosition',
+        ])->getMock();
```

Comment:
`onlyMethods` is deprecated as well. So we swap one deprecated method for another. Not sure that's worth the change.

Reply:
I'm not sure to see where `onlyMethods()` was deprecated? It seems fine to use it according to this:

https://github.com/sebastianbergmann/phpunit/blob/6a17778dc52a6614696c90977dd1162f64a6b5b3/src/Framework/MockObject/MockBuilder.php#L207

Reply:
I mixed up some stuff, my bad. 😓